### ISSUE TITLE
(b/73062247): What we really want is a ReduceWindow with different 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#339]
arrays for index selection vs return value selection--a select-to-gather.

Here, we implement a bitwise hack: we use the hi 16 bits of input for
separate max pooling alongside each of the hi and lo 16 bits of
out_backprop packed into 16 lo bits, which we then glue back together at
the end to get a full 32 bits of gradient.

This could select the wrong backprop value for two x values that are
equally maximal up to the first 16 bits, in which case we are taking the
latter.

Note that in principle we could use 32 separate maxpools to recover each
of 32 bits of the gradient while preserving 31 bits of input for the max
pooling criteria; here, we just truncate to the first 16 bits of input.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [f261257ab26802cf3cab7303a76db2fb729e1d01](https://github.com/imdone/tensorflow/commit/f261257ab26802cf3cab7303a76db2fb729e1d01) by Brian Patton bjp@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/compiler/tf2xla/kernels/pooling_ops.cc#L597-L616
